packages/front-end/components/Experiment/ImportExperimentModal.tsx

```diff
@@ -0,0 +1,91 @@
+import { FC, useEffect, useState } from "react";
+import NewExperimentForm from "./NewExperimentForm";
+import SelectField from "../Forms/SelectField";
+import ImportExperimentList from "./ImportExperimentList";
+import { ExperimentInterfaceStringDates } from "back-end/types/experiment";
+import Modal from "../Modal";
+import { useDefinitions } from "../../services/DefinitionsContext";
+import { useAuth } from "../../services/auth";
+
+const ImportExperimentModal: FC<{
+  onClose: () => void;
+  initialValue?: Partial<ExperimentInterfaceStringDates>;
+  importMode?: boolean;
+  source?: string;
+}> = ({ onClose, initialValue, importMode = true, source }) => {
+  const { datasources } = useDefinitions();
+  const [
+    selected,
+    setSelected,
+  ] = useState<null | Partial<ExperimentInterfaceStringDates>>(initialValue);
+  const [importModal, setImportModal] = useState<boolean>(importMode);
+  const [datasourceId, setDatasourceId] = useState(datasources?.[0]?.id);
+  const [importId, setImportId] = useState(null);
+
+  const { apiCall } = useAuth();
+
+  const getImportId = async () => {
+    const res = await apiCall<{ id: string }>("/experiments/import", {
+      method: "POST",
+      body: JSON.stringify({
+        datasource: datasourceId,
```

Comment:
datasourceId might be undefined.  We shouldn't do an API call in that case.

Reply:
done